### DISCUSSION TITLE
How to redirect stderr to a file in bash?

### DISCUSSION THREAD

**Discussion by jcubic** [oven-sh/bun#8678] [Q&A]
I use this `bun bun.js 2>&1 | less`

and I see this stuff:

![Przechwycenie obrazu ekranu_2024-02-04_13-38-21](https://github.com/oven-sh/bun/assets/280241/f88230c8-79fa-47d9-96e9-0425e8264790)

I want to see the stack trace of the code that Crashes NodeJS (it seems that there is some infinite loop). When I run the code without redirect I see this:

![Przechwycenie obrazu ekranu_2024-02-04_13-39-43](https://github.com/oven-sh/bun/assets/280241/1961926b-6585-4d60-8846-3e80e3bae647)

But there are too much stuff in the console and I don't see the beginning which should show the error message.

**Comment by guest271314** [comment]
Have you tried `head`? You can do something like `bun run bun.js 2>&1 | while read message; do echo "$message" > err.txt; done`.

**Reply by jcubic** [reply]
I used `more` and it gave a lot of blank space at the beginning. So I guess `head` will give the same. Unfortunately, I've fixed the bug in my code and I'm not able to reproduce the bun issue anymore.
